### PR TITLE
ENH: Add t1_bids_path parameter to get_head_mri_trans()

### PR DESCRIPTION
**Pull request by hoechenberger**
The new parameter allows to generate the head <> MRI trans from recordings belonging to different sessions or even separate datasets.

Scenario:

- 1st session: MR
- all following sessions: MEG

In `main`, it wouldn't be possible to use `get_head_mri_trans()` to get the transformation matrix, as MNE-BIDS cannot find the required files.

**RFC**: Should the param maybe be called `t1w_bids_path` instead? (t1w instead of t1)

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/read.py [hunk 1]
```diff
@@ -585,30 +585,38 @@ def read_raw_bids(bids_path, extra_params=None, verbose=True):
...
-        the lankmarks from the MEG file.
-        If a dict, for example: ``extra_params=dict(allow_maxshield=True)``.
+        Extra parameters to be passed to :func:`mne.io.read_raw` when reading
+        the MEG file.
+    t1_bids_path : mne_bids.BIDSPath | None
+        You may optionally specify a :class:`mne_bids.BIDSPath` pointing to the
+        T1-weighted MRI scan that shall be used for extraction of MRI
+        landmarks. Use this parameter if the T1 scan was recorded during a
+        different session than the MEG. It is even possible to point to a
+        T1 image stored in a different **BIDS dataset** than the MEG data.
```
Comment (agramfort) [inline mne_bids/read.py:612]:
```suggestion
    t1_bids_path : mne_bids.BIDSPath | None
        If None, the T1-weighted MRI file will be found based on MEG bids_path.
        You may optionally specify a :class:`mne_bids.BIDSPath` pointing to the
        T1-weighted MRI scan that shall be used for extraction of MRI
        landmarks. Use this parameter if the T1 scan was recorded during a
        different session than the MEG. It is even possible to point to a
        T1 image stored in a different **BIDS dataset** than the MEG data.
```

File: mne_bids/read.py [hunk 2]
```diff
@@ -619,22 +627,28 @@ def get_head_mri_trans(bids_path, extra_params=None):
...
+                                               extension='.json')
+    else:
+        t1_bids_path = t1_bids_path.copy().update(suffix='T1w',
+                                                  datatype='anat')
+        t1w_json_path = _find_matching_sidecar(t1_bids_path, suffix='T1w',
+                                               extension='.json')
 
     # Get MRI landmarks from the JSON sidecar
-    with open(t1w_json_path, 'r', encoding='utf-8-sig') as f:
+    with open(t1w_json_path, 'r', encoding='utf-8') as f:
```
Comment (agramfort) [inline mne_bids/read.py:651]: why did you change this?

**Review by agramfort — APPROVED** [review]
besides LGTM !

File: mne_bids/read.py (hunk 2 above)
Comment (hoechenberger) [reply mne_bids/read.py:654]: Good catch, forgot to mention this!

So it SHOULD have been `utf-8` all along; the UTF BOM character which is handled by `utf-8-sig` is invalid in JSON. Meaning that we shouldn't even try to read it because … well, it shouldn't be present!

One could argue that this change could break backward-compat for some users; however their datasets wouldn't be BIDS-compatible anyway if their JSON files contain a BOM (as, like I said, the BOM is invalid in JSON)

In all other places of MNE-BIDS, we read & write JSON using `utf-8` these days, not `utf-8-sig`; I suppose this line here was just overlooked in #625

**Comment by agramfort** [comment]
@sappelhoff i let you merge

**Review by sappelhoff — APPROVED** [review]
lgtm, thanks :)
